data_safe_haven/functions/miscellaneous.py

```diff
@@ -4,6 +4,16 @@
 import pytz
 
 
+def allowed_dns_lookups() -> list[str]:
+    dns_lookups = {
+        "clamav": ["*.clamav.net", "database.clamav.net.cdn.cloudflare.net"],
+        "oauth": ["login.microsoftonline.com"],
+        "private_dns": [f"*.{zone_name}" for zone_name in ordered_private_dns_zones()],
+        "ubuntu_setup": ["keyserver.ubuntu.com"],
+    }
+    return sorted({zone for zones in dns_lookups.values() for zone in zones})
```

Comment:
They could equally easily be list properties but that feels like a stylistic point. If you want them to be simple lists, that's also possible, but would involve a bit of duplication since one depends on the other.

What change would you like to see?

Reply:
Just taken a look at the code. `ordered_private_dns_zones` takes an argument `resource_type` so you can get the private DNS zones for e.g. just `Azure Automation` if you need. So this one has to be a function. `allowed_dns_lookups` doesn't need to be, but I based it on the previous function. Would you like to see `allowed_dns_lookups` as a plain list instead?

Reply:
That sounds sensible, clearer than lots of dict slices over the code.

Happy to leave this as it is.